apps/nextjs/src/hooks/useLessonPlanScrollManagement.ts

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect, useRef } from "react";
```

Comment:
Our patterns of useEffects and inline hooks in the render function make this hard for me to follow without understanding every line. Typically, we could work on it and find a clearer pattern, but this PR is so bit that it feels irresponsible to make a suggestion that would delay it any longer